d2core/d2object/init_function.go

```diff
@@ -27,15 +27,15 @@ func initObject(ob *Object) bool {
 
 // Initializes torch/brazier type objects
 func initTorch(ob *Object) {
-	if ob.objectRecord.HasAnimationMode[d2enum.AnimationModeObjectOperating] {
-		ob.setMode("ON", 0, true)
+	if ob.objectRecord.HasAnimationMode[d2enum.ObjectAnimationModeOperating] {
+		ob.setMode(d2enum.ObjectAnimationModeOpened, 0, true)
```

Comment:
Oops, I guess I messed up those checks when I implemented those.

They should check for and set ObjectAnimationModeOpened